src/compiler/config/validate-dev-server.ts

```diff
@@ -100,7 +100,7 @@ export const validateDevServer = (
     devServer.protocol = devServer.https ? 'https' : addressProtocol ? addressProtocol : 'http';
   }
 
-  if (devServer.historyApiFallback !== null && devServer.historyApiFallback !== false) {
```

Comment:
Feel free to push back on this.

Reply:
This logic actually is the validation on the field 😉 

I agree, this may cause issues. However, we can't coerce `historyApiFallback` to a boolean with a double-bang operator, because that unfortunately changes the logic of the conditional as well; instead of  running the consequence of the if-statement whenever `historyApiFallback` is not `false` (prior to my proposed changed) to running the consequence of the if-statement whenever `historyApiFallback` is `true` - since this is an untyped field, that means that if `historyApiFallback === 1` for instance, we've changed the fallback logic.

In my view, it comes down to how many guard rails we want to put up for folks. With my proposed change, we take some down in the name of type safety. We could also do something like this:
```diff
- if (devServer.historyApiFallback !== null && devServer.historyApiFallback !== false) {
+ // this field could be anything, let's only act on it if and only if it's truthy
+ if (devServer.historyApiFallback !== null && (devServer.historyApiFallback as unknown as boolean) !== false) {
```
which keeps the logic exactly the same, but is still weird if `devServer.historyApiFallback === 1` as soon as we enter the consequence:
```ts
    // this is 1 (that doesn't seem right),
    devServer.historyApiFallback = devServer.historyApiFallback || {};

    if (!isString(devServer.historyApiFallback.index)) {
      devServer.historyApiFallback.index = 'index.html';
    }
```

I think there are a few subtle bugs in this logic, that's "break" a config if it's not typed properly. Perhaps the best thing to do is if `historyApiFallback` is not null, coerce it to an object literal if it's not one already:
```diff
   if (devServer.historyApiFallback !== null) {
-    devServer.historyApiFallback = devServer.historyApiFallback || {};
+    if (Array.isArray(devServer.historyApiFallback) || typeof devServer.historyApiFallback !== 'object') {
+      devServer.historyApiFallback = {};
+    }

```

Reply:
Thoughts?

Reply:
I kinda like that. We can't protect against _everything_ a user might do in their config, but having some type checks are an easy thing we can do.